pgmpy/models/BayesianNetwork.py

```diff
@@ -1364,7 +1363,51 @@ def simulate(
                 model.add_cpds(new_cpd)
                 evidence[new_var] = 0
 
-        # Step 3: If no evidence do a forward sampling
+        # Step 3: If missing_prob; include missing values in samples.
+        if missing_prob:
+            if isinstance(missing_prob, list):
+                for cpd in missing_prob:
+                    if not isinstance(cpd, TabularCPD):
+                        raise ValueError(
+                            "Missing probability should only contain TabularCPD."
```

Comment:
The error message can be a bit more precise. Something like `f"missing_prob must be a list of TabularCPD objects. Got {type(cpd)}"`